### ISSUE TITLE
Non homogenous behaviour of LED in exam mode

### ISSUE THREAD

**Issue by Overengined** [UpsilonNumworks/Upsilon#316]
**Describe the bug**
Exam mode LED is bahving differently depending on whether the device is on or off.

**To Reproduce**
1. Put the calculator in exam mode and take notice of the LED blink speed.
2. Turn the calculator off and notice that the LED is blinking at a different speed.

**Expected behavior**
The LED should blink at the same speed, otherwise an over-zealous examinator might find it supsicious.

**Device (please complete the following information):**
 - N0110
 - Upsilon Version: 1.0.1 -dev
 - Upsilon commit: 4acc79f
 (seems to be version independant tho)

**Comment by 0x4c756e61** [comment]
I was able to reproduce it, and it looks like it is slower when we turn off the calculator

**Comment by Overengined** [comment]
Can confirm

**Comment by theodorechle** [comment]
Same thing when it does something (calculation, execution of a program, ...). It's faster than when it does nothing.

**Comment by Overengined** [comment]
In my opinion this issue should get tagged in a way that illustrates its importance. I'm not saying we're dealing with something major here, but since it's related to the exam mode, it deserves some attention.

**Comment by NowanPython** [comment]
I just checked the exam mode and I noticed that when using ΧCAS' `plot3d` it blinks so fast that anyone would probably find it suspicious.

**Comment by Overengined** [comment]
@Lauryy06 would it be possible to kncrease the priority of this issue ? We dont have ans idea why its happening. My best guess would be that the blinking is set with cpu cycles and so when they get longer the blinking gets slower. Opposite works too.

**Comment by Lisra-git** [comment]
Hard to fix this issue, it exist since Omega and was only "fixed" partially in Epsilon recently.

**Comment by Overengined** [comment]
> Epsilon

How did tjey do it ?

**Comment by Overengined** [comment]
> Hard to fix this issue, it exist since Omega and was only "fixed" partially in Epsilon recently.

How did they do it ?

**Comment by GustavoSchip** [comment]
> > Hard to fix this issue, it exist since Omega and was only "fixed" partially in Epsilon recently.
> 
> How did they do it ?

Exactly, I will try and see what they changed. Hope that change isn't too big or complicated.

**Comment by Lisra-git** [comment]
Surprise, source is private

**Comment by GustavoSchip** [comment]
> Surprise, source is private

Huh? I thought they still had the repo updated, perhaps I am mistaken 😭

**Comment by Overengined** [comment]
The Epsilon OS isn't private, but it has gone from free software to closed source. So you legally cannot copy anything they do in their code.

**Comment by AngeDieu** [comment]
but can you get inspired by their code and then make a other similar one, or not similar ?

**Comment by GustavoSchip** [comment]
> but can you get inspired by their code and then make a other similar one, or not similar ?

Yes you can, as long as it is not a direct copy, but will probably remain in a grey area

**Comment by GustavoSchip** [comment]
> The Epsilon OS isn't private, but it has gone from free software to closed source. So you legally cannot copy anything they do in their code.  

Then we will just observe how they fixed it, and implement it similarly or differently, as long as the issue is fixed and nothing illegal has been done

**Comment by Lisra-git** [comment]
The relevant code is private

**Comment by GustavoSchip** [comment]
> The relevant code is private

Seriously...
Then we would have to look at this issue ourselves.
Once I am back home from my vacation I'll try and see what is up with this

**Comment by shipped-it** [comment]
Any updates on this? It's still an issue.
